openproblems/tasks/spatial_decomposition/methods/rctd.R

```diff
@@ -0,0 +1,48 @@
+library(RCTD)
+library(SingleCellExperiment)
+library(Matrix)
+
+# R base for rctd.py
+
+# extract single cell reference data
+sce_sc <- sce[, colData(sce)$modality == "sc"]
+# get single cell reference counts
+sc_counts <- assay(sce_sc, "X")
+# get single cell reference labels
+sc_cell_types <- factor(colData(sce_sc)$label)
+names(sc_cell_types) <- colnames(sce_sc)
+
+# construct reference object (specific for RCTD)
+reference <- Reference(sc_counts, sc_cell_types)
+# extract spatial data
+sce_sp <- sce[, colData(sce)$modality == "sp"]
+# get spatial data counts
+sp_counts <- assay(sce_sp, "X")
+# get spatial data coordinates
+sp_coords <- as.data.frame(sce_sp@reducedDims$spatial)
+colnames(sp_coords) <- c("x", "y")
+rownames(sp_coords) <- colnames(sce_sp)
+rownames(sp_counts) <- rownames(sce_sp)
+colnames(sp_counts) <- colnames(sce_sp)
+# create spatial object to use in RCTD
+puck <- SpatialRNA(sp_coords, sp_counts)
+# create RCTD object from reference and spatialRNA objects
+my_rctd <- create.RCTD(puck, reference, max_cores = 1, test_mode = FALSE)
```

Comment:
shall we set an higher number of cores? just asked on `infrastructure if that's allowed`

Reply:
Awesome, thanks for asking - let's wait and see what they say; I was as restrictive as possible here - but would be great if we could set more.